config/routes.rb

```diff
@@ -39,7 +39,10 @@
   devise_scope :user do
     get "/users/sign_up" => redirect("/users/sign_in")
   end
-  devise_for :users, :controllers => {:registrations => "users/registrations"}
+  devise_for :users, :controllers => {
+      :registrations => "users/registrations",
+      :omniauth_callbacks => "users/omniauth_callbacks"
```

Comment:
They tell Devise to use our controllers (inside `app/controllers`) for `:registrations` and `:omniauth_callbacks`.